src/test/java/spark/route/RouteEntryTest.java

```diff
@@ -106,14 +95,61 @@ public void testMatches_PathsDoNotMatch() {
 
         RouteEntry entry = new RouteEntry();
         entry.httpMethod = HttpMethod.get;
+
+        entry.path = "/test/*";
+
+        assertFalse("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/"));
+
+        entry.path = "/test/me/:id";
+
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid"));
+
+        entry.path = "/test/me/:id/*";
+
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid/child"));
+
         entry.path = "/test/this/resource/*";
 
         assertTrue("Should return true because path specified is covered by the route path wildcard",
-                   entry.matches(HttpMethod.get, "/test/this/resource/child/id"));
+            entry.matches(HttpMethod.get, "/test/this/resource/child/id"));
+
+        entry.path = "/test/this/*/resource/*";
+
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/this/a/resource/child/id"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/this/a/another/resource/child/id"));
+
+        entry.path = "/test/me/:id/this/*/resource/*";
+
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid/this/a/resource/child/id"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid/this/a/another/resource/child/id"));
+
+        entry.path = "/test/me/:id/*/resource/*";
+
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid/a/resource/child/id"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/me/myid/a/another/resource/child/id"));
+
+        entry.path = "/test/this/*/resource";
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
+            entry.matches(HttpMethod.get, "/test/this/a/resource"));
+        assertTrue("Should return true because path specified is covered by the route path wildcard",
```

Comment:
any clue what happens with /test/this/a/resource/a/resource. Is this a greedy match?

Reply:
it matches the pattern, so it'll match